src/zenml/integrations/kubernetes/orchestrators/kubernetes_orchestrator_entrypoint.py

```diff
@@ -174,6 +192,20 @@ def run_step_on_kubernetes(step_name: str) -> None:
 
     logger.info("Orchestration pod completed.")
 
+    if (
```

Comment:
I only run this cleanup in case the orchestration pod exited succesfully. This is in order to prevent accidently deleting the secret while a step pod is still running, but will lead to some secrets being left over if the pipeline failed. @stefannica what do you think about this?

Reply:
If I'm reading the `ThreadedDagRunner` implementation correctly, this code is only reached when all of the pods have completed running (i.e. all of the threads have been "joined"). So you don't need this condition at all.

Reply:
Which condition? I'm not explicitly checking for success anywhere, but implicitly by putting the code where it is right now

Reply:
Oh, I see... then why don't you just use a try/finally around `ThreadedDagRunner.run()` ?

Reply:
That is precisely my question: This would handle also the case where a step is failing. There is however one other option that I've also seen sometimes: The dag runner fails with some random error, even though the step is actually still running. In that case, we would remove the secret already. Now that I'm typing it however, I realize that at this point the token would already be set as environment variable, which means we can delete it without disturbing the (potentially still active) step. I'll adjust it.